Add option in `RustConfig` to customize the build.

yageek opened the pull request:
Adds the `CARGO_BUILD_FLAGS` option to be able to customize the build command.

yageek committed:
Add option in `RustConfig` to customize the build.

emk commented on bin/compile:
```diff
@@ -32,6 +32,8 @@ BUILD_PATH=""
 # These flags are passed to `cargo install diesel`, e.g. '--no-default-features --features postgres'
 DIESEL_FLAGS=""
+# `CARGO_BUILD_FLAGS` default build command
+CARGO_BUILD_FLAGS=""
```
We definitely want to build with `--release` by default as before. So as a minimum, we'd need:

```suggestion
RUST_CARGO_BUILD_FLAGS="--release"
```

Also, I'd like to name this `RUST_CARGO_BUILD_FLAGS` by default, because that will make things clearer in multi-build-pack situations. Yes, some of the existing flags don't have a `RUST_*` prefix. I plan to fix that some day.

yageek replied:
Oupss. I forgot to commit my last changes... I'll update it

emk commented on bin/compile:
```diff
@@ -135,7 +137,7 @@ if [ $RUST_SKIP_BUILD -ne 1 ]; then
   cd "$BUILD_DIR/$BUILD_PATH"
   rm -rf target/
-  cargo build --release
+  cargo build "$CARGO_BUILD_FLAGS"
```
```suggestion
  cargo build $RUST_CARGO_BUILD_FLAGS
```
You definitely don't want the `"` characters around `$RUST_CARGO_BUILD_FLAGS` here, because that will cause the shell to consider all the flags to be one, single command-line argument.

emk commented on bin/compile:
```diff
@@ -32,6 +32,8 @@ BUILD_PATH=""
 RUST_INSTALL_DIESEL=0
 # These flags are passed to `cargo install diesel`, e.g. '--no-default-features --features postgres'
 DIESEL_FLAGS=""
+# `CARGO_BUILD_FLAGS` default build command
```
```suggestion
# Default build flags to pass to `cargo build`.
```

emk commented on README.md:
```diff
@@ -98,6 +98,10 @@ But if you only need Rust to build a particular Ruby gem, and you have no top-le
 
+## Customizing build flags
+
+By default the command line used to build the rust binary is `cargo build --release`.
```
This help message should explain how to actually set the build flags, and what the default value of `RUST_CARGO_BUILD_FLAGS` is. This will help users figure out how to customize it!

emk reviewed (changes requested):
Thank you for submitting this PR! I have a number of suggestions below about how we could tweak this to be as useful to as many people as possible. Thank you for considering them!

yageek committed:
Implements requires changes.

- Rename the variable comment
- Remove double quotes for the `cargo buil` command.
- Details documention in the README.md

emk commented:
Thank you for your change, and my apologies for taking so long to follow up on it!
